Src/library/net/Clib/network.c

```diff
@@ -1176,8 +1176,28 @@ EIF_INTEGER c_set_sock_opt_linger(EIF_INTEGER fd, EIF_BOOLEAN flag, EIF_INTEGER
 	return (EIF_INTEGER) result;
 }
 
-EIF_INTEGER c_get_sock_recv_timeout(EIF_INTEGER fd, EIF_INTEGER level)
-	/* get socket fd SO_RCVTIMEO options */
+/* common timeval sockopt functions */
+
+EIF_INTEGER c_sockopt_timeval_ns_to_seconds(EIF_INTEGER_64 ns) 
+	/* truncate nanoseconds in seconds */
+	/* note: if ns is above zero, then return 1 second instead of zero to avoid eventual special behavior for 0 timeouts */
+{
+	EIF_INTEGER secs;
+	if (ns == 0) {
+		secs = 0;
+	} else {
+		secs = (EIF_INTEGER) (ns / RTI64C(1000000000)); /* Convert nanoseconds to seconds */
```

Comment:
I gave a second thought on the logic and wonder if the calculation (here and at other places) could be simplified to
```
   us = (ns + 999) / 1_000
   ms = (ns + 999_999) / 1_000_000
   s = (ns + 999_999_999) / 1_000_000_000
```
This would be faster and also it would ensure the timeout is the specified value or a bit longer, but never shorter.